tests/integration/new_relations/test_new_relations.py

```diff
@@ -295,3 +295,66 @@ async def test_an_application_can_request_multiple_databases(ops_test: OpsTest,
 
     # Assert the two application have different relation (connection) data.
     assert first_database_connection_string != second_database_connection_string
+
+
+async def test_relation_data_is_updated_correctly_when_scaling(ops_test: OpsTest):
+    """Test that relation data, like connection data, is updated correctly when scaling."""
+    # Retrieve the list of current database unit names.
+    units_to_remove = [unit.name for unit in ops_test.model.applications[DATABASE_APP_NAME].units]
+
+    # Add two more units.
+    await ops_test.model.applications[DATABASE_APP_NAME].add_units(2)
+    await ops_test.model.wait_for_idle(
+        apps=[DATABASE_APP_NAME], status="active", timeout=1000, wait_for_exact_units=4
+    )
```

Comment:
`ops_test.fast_forward` might help here

Reply:
Good suggestion. Added in https://github.com/canonical/postgresql-operator/pull/17/commits/391aaa27327ff9d768c2ca34b336fd6695b6fcd0.